src/net_permissions.h

```diff
@@ -51,8 +51,14 @@ class NetPermissions
     {
         flags = static_cast<NetPermissionFlags>(flags | f);
     }
+    //! ClearFlag() is only used with NetPermissionFlags::PF_ISIMPLICIT. If that
+    //! is changed in the future, in its current version ClearFlag() should not
+    //! be called with subflags of multiflags, e.g. NetPermissionFlags::PF_RELAY
+    //! or NetPermissionFlags::PF_DOWNLOAD, as that would leave the result in an
+    //! invalid state corresponding to none of the existing flags.
     static inline void ClearFlag(NetPermissionFlags& flags, NetPermissionFlags f)
     {
+        assert(f == NetPermissionFlags::PF_ISIMPLICIT);
```

Comment:
During fuzzing `ClearFlag()` is called with a fuzzed/random argument:

https://github.com/bitcoin/bitcoin/blob/773f8c1a7d568012768e16d2ede65b5d4d62aced/src/test/fuzz/net_permissions.cpp#L24-L39

I guess this means that eventually this newly added assert will be triggered? I think either remove the assert or remove the `ClearFlag()` calls from the fuzz tests.

Reply:
Ah, didn't check the fuzzer. Thanks!

Reply:
Fixed.